### ISSUE TITLE
Area ridges: fill overlaps geometry borders when colors are repeated

### ISSUE THREAD

**Issue by ASmirnov-HORIS** [JetBrains/lets-plot#674]
Code:

```python
data = {
    "x": [0, 0, 1, 1, 2, 2, 3, 3, 4],
    "y": [0, 0, 0, 0, 0, 0, 0, 0, 0],
    "h": [1, 1, .8, .8, .2, .2, .8, .8, 1],
    "q": [0, 0, 0, .5, .5, .8, .8, 1, 1],
}
ggplot(data, aes("x", "y")) + \
    geom_area_ridges(aes(height="h", quantile="q", fill="q"), stat='identity') + \
    scale_fill_viridis(option='twilight')
```

Result:

<img src="https://user-images.githubusercontent.com/26162403/215742331-d3776615-1616-460f-9c06-d80bf8063655.png" width="400"></img>

**Comment by ASmirnov-HORIS** [comment]
The same problem arises for the following scale:

```python
scale_fill_gradientn(colors=["red", "blue", "red"])
```

I think it's necessary for colors to repeats.

**Comment by alshan** [comment]
Looks like an expected behavior. Why it is a bug in your opinion?

**Comment by ASmirnov-HORIS** [comment]
This is an expected behaviour:

<img src="https://user-images.githubusercontent.com/26162403/221833129-368e2aa1-26be-4166-ae1d-cac778fdde0f.png" width="400"></img>

It looks the same for all scales that have no repeated colors, i.e. for all but `scale_fill_viridis(option='twilight')` and `scale_fill_gradientn(colors=%special_bad_values%)`

**Comment by alshan** [comment]
But viridis twilight colormap starts and ends with the same "off-white" color by design:

> The colors begin with off-white, ranges through blue to off-black, then to red and back to off-white.

https://github.com/bastibe/twilight

**Comment by ASmirnov-HORIS** [comment]
I apologize if my wording was misleading. Viridis is not directly related to the problem.

The problem itself is that the fill is out of geometry, and it shows up when the colors are repeated. Here are 2 examples with `gradientn` scale:

```python
data = {
    "x": [0, 0, 1, 1, 2, 2, 3, 3, 4],
    "y": [0, 0, 0, 0, 0, 0, 0, 0, 0],
    "h": [1, 1, .8, .8, .2, .2, .8, .8, 1],
    "q": [0, 0, 0, .5, .5, .8, .8, 1, 1],
}
ggplot(data, aes("x", "y")) + \
    geom_area_ridges(aes(height="h", quantile="q", fill="q"), stat='identity') + \
    scale_fill_gradientn(colors=["#0000ff", "#ff0000", "#0000ff"])
```

<img src="https://user-images.githubusercontent.com/26162403/222398846-fe6d70e3-2d06-48fc-a410-cc069bb9d877.png" width="400"></img>

```python
ggplot(data, aes("x", "y")) + \
    geom_area_ridges(aes(height="h", quantile="q", fill="q"), stat='identity') + \
    scale_fill_gradientn(colors=["#0000ff", "#ff0000", "#0100ff"])
```

<img src="https://user-images.githubusercontent.com/26162403/222399041-3497ebb4-548a-4539-9ae6-1864caee55ea.png" width="400"></img>

A minimal difference in `colors` parameter - and the extra part of the fill is gone. In my opinion this doesn't look like the expected behavior.

**Comment by ASmirnov-HORIS** [comment]
Fixed in #707.
